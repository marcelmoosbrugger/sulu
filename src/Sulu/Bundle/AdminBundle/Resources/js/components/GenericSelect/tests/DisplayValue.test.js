/* eslint-disable flowtype/require-valid-file-annotation */
import {render, shallow} from 'enzyme';
import React from 'react';
import DisplayValue from '../DisplayValue';

test('The component should render', () => {
    const displayValue = render(<DisplayValue>My value</DisplayValue>);
    expect(displayValue).toMatchSnapshot();
});

test('The component should render with an icon', () => {
    const displayValue = render(<DisplayValue icon="plus">My value</DisplayValue>);
    expect(displayValue).toMatchSnapshot();
});

test('A click on the component should fire the callback', () => {
    const clickSpy = jest.fn();
    const displayValue = shallow(<DisplayValue onClick={clickSpy}>My value</DisplayValue>);
    displayValue.simulate('click');
    expect(clickSpy).toBeCalled();
});
