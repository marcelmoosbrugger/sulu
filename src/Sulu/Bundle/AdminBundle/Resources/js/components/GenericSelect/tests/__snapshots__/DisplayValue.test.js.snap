// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render 1`] = `
<button
  class="displayValue"
>
  My value
  <span
    aria-hidden="true"
    class="toggle fa fa-chevron-down"
  />
</button>
`;

exports[`The component should render with an icon 1`] = `
<button
  class="displayValue hasIcon"
>
  <span
    aria-hidden="true"
    class="frontIcon fa fa-plus"
  />
  My value
  <span
    aria-hidden="true"
    class="toggle fa fa-chevron-down"
  />
</button>
`;
